src/server/routes/auth.js

```diff
@@ -214,32 +225,37 @@ router.get('/scanners', helpers.ensureAdmin, function (req, res) {
  *       message: "There was an error."
  *     }
  */
-router.post('/scanner/register', function (req, res, next) {
-  if (!req.body.pin || req.body.pin !== process.env.SCANNER_ADMIN_PIN) {
+router.post('/scanner/register', asyncMiddleware(async function (req, res, next) {
+  if (!req.body.pin) {
     console.error("Invalid pin passed");
-    return res.status(401).send(new Error("Invalid pin passed"));
+    let err = new Error("Invalid pin passed");
+    err.status = 401;
+    return next(err);
   }
-  let pin = req.body.pin;
-  let newScanner = new Scanner();
-  newScanner.apikey = uuidv4();
-  newScanner.name = (new Date()).toISOString();
-  newScanner.save(function (err, results) {
-    if (err) {
-      console.log(err);
-      res.status(500).json({
-        status: "error",
-        data: err,
-        message: "There was an error."
-      });
-    } else {
-      res.status(200).json({
-        status: "success",
-        data: results,
-        message: "Scanner Added. API Key Generated."
-      });
-    }
-  });
-});
+  let scanner = await Scanner.findOne({pin: req.body.pin}).exec();
```

Comment:
```suggestion
  let scanner = await Scanner.findOne({ pin: req.body.pin }).exec();
``` for ESLint compliance?